scripts/tests/sanitycheck/test_testsuite_class.py

```diff
@@ -236,13 +236,22 @@ def test_apply_filters_part2(class_testsuite, all_testcases_dict,
     Part 2 : Response of apply_filters function (discard dictionary) have
              appropriate values according to the filters
     """
+
     class_testsuite.platforms = platforms_list
     class_testsuite.testcases = all_testcases_dict
-    kwargs = {extra_filter : extra_filter_value,
-              "exclude_platform" : ['demo_board_1'], "platform" : ['demo_board_2']}
+    kwargs = {
+        extra_filter : extra_filter_value,
+        "exclude_platform" : [
+            'demo_board_1'
+            ],
+        "platform" : [
+            'demo_board_2'
+            ]
+        }
     discards = class_testsuite.apply_filters(**kwargs)
-    assert type(list(discards.keys())[0]).__name__ == "TestInstance"
-    assert list(dict.fromkeys(discards.values())) == [expected_discards]
```

Comment:
Any reason why the assert on expected discard values is removed? The TESTDATA_PART2 is specifically meant for asserting on the values based on the filters applied.

Reply:
it is replaced by the the for loop that follows to make readable and debuggable.

Reply:
Makes sense, thanks for clarifying this.